workbench/src/renderer/components/LogTab/index.jsx

```diff
@@ -85,9 +86,15 @@ export default class LogTab extends React.Component {
   }
 
   updateState() {
-    this.setState((state) => ({
-      logdata: state.logdata.concat(this.cache)
-    }));
+    // flushSync will override react18 batched updates
+    // and force state updates to happen now. We're managing
+    // the rate of updates ourselves in this component via
+    // debouncedLogUpdate.
+    ReactDom.flushSync(() => {
+      this.setState((state) => ({
+        logdata: state.logdata.concat(this.cache)
+      }));
+    });

```

Comment:
I tested removing the `cache` and `debounced` updates, to just let react manage state updates entirely on its own. That seemed to work fine now, as of React18. But still seemed like there were excessive re-renders during high volume stdout coming from invest.

So I think our approach here is likely safer on machines with lower performance. There's no reason to update the logfile more rapidly than once every 10ms.